main_test.go

```diff
@@ -402,22 +412,42 @@ func expandEscapedFilenames(t *testing.T) error {
 			return string([]byte{byte(code)})
 		})
 		if file != input {
-			if err := os.Link(input, file); err != nil {
+			link := filepath.Join(dstDir, file[9:]) // Strip "testdata/" prefix.
```

Comment:
This part was already quite hacky. Perhaps we can make it simpler by always making a symbolic link or even a copy of the file instead?

Reply:
Funny, I did start writing a version of this that copies the files, and bumped into that unusual lack of a "copy file" procedure in the Go standard library. I went off reading about the various ways to copy files, shook my head, then went back to making the links work.

Since these files are all rather small, copying the files "the easy way" ([`ioutil.ReadFile`](https://golang.org/pkg/io/ioutil/#ReadFile) to [`ioutil.WriteFile`](https://golang.org/pkg/io/ioutil/#WriteFile)) should work well enough.

That wasn't the only challenge, though. The golden output files with error messages inside contain an expected file path. Making the actual output match that closely enough required yet more munging.